store/sqlstore/post_store.go

```diff
@@ -770,7 +770,10 @@ func (s *SqlPostStore) getPostIdAroundTime(channelId string, time int64, before
 			sq.Eq{"ChannelId": channelId},
 			sq.Eq{"DeleteAt": int(0)},
 		}).
-		OrderBy("CreateAt " + sort).
+		// Adding ChannelId and DeleteAt order columns
+		// to let mysql choose the "idx_posts_channel_id_delete_at_create_at" index always.
+		// See MM-23369.
+		OrderBy("ChannelId, DeleteAt, CreateAt " + sort).
```

Comment:
nit: might we write this as follows to minimize the "manual SQL" and maximize Squirrel's flexibility?
```suggestion
		OrderBy("ChannelId", "DeleteAt", "CreateAt " + sort).
```

Reply:
Done